pkg/apis/experimental/types.go

```diff
@@ -495,11 +496,13 @@ type IngressList struct {
 
 // IngressSpec describes the Ingress the user wishes to exist.
 type IngressSpec struct {
-	// TODO: Add the ability to specify load-balancer IP just like what Service has already done?
-	// A list of rules used to configure the Ingress.
-	// http://<host>:<port>/<path>?<searchpart> -> IngressBackend
-	// Where parts of the url conform to RFC 1738.
-	Rules []IngressRule `json:"rules"`
+	// A default backend capable of servicing requests that don't match any
+	// IngressHost. It is optional to allow the loadbalancer controller or
+	// defaulting logic to specify a global default.
+	Backend *IngressBackend `json:"backend,omitempty"`
+	// A list of host rules used to configure the Ingress.
+	Rules []IngressHost `json:"rules"`
```

Comment:
I know you explained why, but making the type Host and the name Rule bother me,  What if this was `Rules []IngressRule` and what you currently call IngressRule -> IngressRuleValue ?